teachertool/src/services/backendRequests.ts

```diff
@@ -88,3 +88,30 @@ export async function loadTestableCollectionFromDocsAsync<T>(fileNames: string[]
 
     return allResults;
 }
+
+export async function askCopilotQuestion(shareId: string, question: string): Promise<string | undefined> {
+    const { state: teacherTool } = stateAndDispatch();
+
+    const url = `${
+        teacherTool.copilotEndpointOverride ? teacherTool.copilotEndpointOverride : pxt.Cloud.apiRoot
+    }copilot/question`;
+
+    const data = { id: shareId, question };
+    let result: string = "";
+    try {
+        const request = await fetch(url, {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify(data),
+        });
+
+        if (!request.ok) {
+            throw new Error("Unable to reach Copilot");
+        }
+        result = await request.json();
+    } catch (e) {
+        logError(ErrorCode.askCopilotQuestion, e);
+    }
+
+    return result;
```

Comment:
Is there any way to indicate progress with the thinking? Because the other evaluations are so fast, I found myself impatient as I waited. Not something needed for this PR and I don't feel strongly about it, but might be something interesting to investigate.

Reply:
Not at present. Deep Prompt just tells us if it's "in progress" or not. We could see if streaming the response text as it comes in or if getting some kind of % completion is in their plans, but I don't think it's possible currently.

Reply:
Sounds good. It would be interesting to find out if that's something they plan to do, but it can probably be something we inquire about after demoing.